server/src/models/io/trainersIo.ts

```diff
@@ -0,0 +1,19 @@
+import { Users, Trainers } from "../db/types"
+
+export type TrainersData = Omit<Users, "user_id" | "password"> & Trainers
+
+export type TrainersApiRequest =    Omit<Users, "user_id" | "type"> &
+                                    Omit<Trainers, "trainer_id"> & 
```

Comment:
I would keep `trainer_id`.
Its not generated so you don't have to Omit it.

You can just change it to `Trainers`